keanu-project/src/test/java/io/improbable/keanu/tensor/dbl/DoubleTensorTest.java

```diff
@@ -212,22 +396,21 @@ public void canSetToZeroWhenNaN() {
 
     @Test
     public void cannotSetIfMaskLengthIsSmallerThanTensorLength() {
-        DoubleTensor tensor = Nd4jDoubleTensor.create(new double[]{1., 2., 3., 4.}, new long[]{2, 2});
-        DoubleTensor mask = Nd4jDoubleTensor.scalar(1.);
+        DoubleTensor tensor = DoubleTensor.create(new double[]{1., 2., 3., 4.}, new long[]{2, 2});
+        DoubleTensor mask = DoubleTensor.scalar(1.);
 
         thrown.expect(IllegalArgumentException.class);
-        thrown.expectMessage("The lengths of the tensor and mask must match, but got tensor length: " + tensor.getLength() + ", mask length: " + mask.getLength());
 
         tensor.setWithMaskInPlace(mask, -2.0);
     }
 
     @Test
     public void cannotSetIfMaskLengthIsLargerThanTensorLength() {
-        DoubleTensor tensor = Nd4jDoubleTensor.scalar(3);
-        DoubleTensor mask = Nd4jDoubleTensor.ones(2, 2);
+        DoubleTensor tensor = DoubleTensor.create(3);
+        DoubleTensor mask = DoubleTensor.ones(2, 2);
 
         thrown.expect(IllegalArgumentException.class);
-        thrown.expectMessage("The lengths of the tensor and mask must match, but got tensor length: " + tensor.getLength() + ", mask length: " + mask.getLength());
+//        thrown.expectMessage("The lengths of the tensor and mask must match, but got tensor length: " + tensor.getLength() + ", mask length: " + mask.getLength());
```

Comment:
delete if this isn't expected

Reply:
I've fixed the Tensor code to return the correct error message now.